### ISSUE TITLE
 localhost could not be resolved (3: Host not found), client: 127.0.0.1, server: localhost, request: "GET / HTTP/1.1", host: "localhost"

### ISSUE THREAD

**Issue by theret** [engintron/engintron#303]
Hey

Engintron is king. Love it ;) 

I have only one error that keeps showing every 15 seconds or so : 

**2016/11/03 17:31:17 [error] 788264#788264: *51970 localhost could not be resolved (3: Host not found), client: 127.0.0.1, server: localhost, request: "GET / HTTP/1.1", host: "localhost"
2016/11/03 17:31:34 [error] 788264#788264: *51975 localhost could not be resolved (3: Host not found), client: 127.0.0.1, server: localhost, request: "GET / HTTP/1.1", host: "localhost"**

Any ideas?

Thanks

**Comment by fevangelou** [comment]
Make sure there's an entry for "localhost" in the same line with "127.0.0.1 ", in your /etc/hosts file.

**Comment by theret** [comment]
Hey
I do have an entry : 

![hosts](https://cloud.githubusercontent.com/assets/13104846/19999188/c04a6d9e-a268-11e6-83b1-b6e577803ef4.jpg)

**Comment by theret** [comment]
[SOLVED]

I know it's strange but actually removing the line 
127.0.0.1    localhost 

from /etc/hosts solved the issue for me...

**Comment by fevangelou** [comment]
You should leave that entry there.

**Comment by revcomllc** [comment]
we have had the same issues @theret . Yet to find a solution

**Comment by fevangelou** [comment]
Try this please. Edit your "custom rules" file via Engintron's WHM app and at the bottom, add this block

```
if ($host ~ "localhost") {
    set $PROXY_DOMAIN_OR_IP "127.0.0.1";
}
```

**Comment by revcomllc** [comment]
@fevangelou  I believe this solution worked!

**Comment by fevangelou** [comment]
Cool, I'm gonna merge it in v1.8.0 - remember to remove it from your "Custom Rules" file when the new release is out.

**Comment by theret** [comment]
Thanks for the solution - works ;)

**Comment by sterlingken** [comment]
Worked for me too! Thanks so much @fevangelou

**Comment by fevangelou** [comment]
Closing issue and moving to wiki until merged with Engintron.

**Comment by fevangelou** [comment]
https://github.com/engintron/engintron/wiki#troubleshooting

**Comment by CarlosLopezES** [comment]
Hi,

Sorry for the brutal update from old post, but i'm facing this same problem and Google send me to here!

Since this morning, i have this error:

2018/06/08 12:51:17 [error] *426 www.XXX.es could not be resolved (3: Host not found), client: 85.25.210.41, server: localhost, request: "GET /robots.txt HTTP/1.1", host: "www.XXX.es"

Only ocurrs on some domanins and not every time, but maybe the 80% of the tries.

No software change, until today all was working fine. I'm not using cloudlfare or anything like that.

Thank you!
